### ISSUE TITLE
Can't resolve 'react-query/persistQueryClient'

### ISSUE THREAD

**Issue by wottpal** [tmm/wagmi#724]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.5.9

### Current Behavior

If I install dependencies with `pnpm` and try to run the dev environment it throws the following error:

```
error - ../../node_modules/.pnpm/wagmi@0.5.9_w3nw2bpaojq3mlwhwp6lup3gdy/node_modules/wagmi/dist/wagmi.esm.js:5:0
Module not found: Can't resolve 'react-query/persistQueryClient'

Import trace for requested module:
./src/pages/_app.tsx

https://nextjs.org/docs/messages/module-not-found
```

Before, it didn't do that and just worked fine. 

I just figured out that 30min ago [`react-query@4.0.0`](https://www.npmjs.com/package/react-query) was released on npm. I guess this correlates somehow?

**Comment by oveddan** [comment]
Happens for me too when using yarn - whats the approach to fix there?

**Comment by jxom** [comment]
Working on a fix.

**Comment by oveddan** [comment]
Thanks for the suggestion, I'm still getting the same issue...

`package.json`:

```
  "resolutions": {
    "react-query": "4.0.0-beta.23"
  },
```

My yarn.lock now looks like:
```
react-query@^4.0.0-beta.23:
  version "4.0.0"
  resolved "https://registry.yarnpkg.com/react-query/-/react-query-4.0.0.tgz#d8a5347503d0aedebc957e35b408c4aa3f726580"
  integrity sha512-qiW+Yvbl+EK8iwPDJAj4qWAKceh+g8Up8jxoNxJbzhV3bNheeyHF3EyynnkDO3S+CYgSwtCUFaP8vOjB62j7xQ==
  dependencies:
    "@tanstack/query-core" "^4.0.0-beta.1"
    "@types/use-sync-external-store" "^0.0.3"
    use-sync-external-store "^1.2.0"


wagmi@^0.5.9:
  version "0.5.9"
  resolved "https://registry.yarnpkg.com/wagmi/-/wagmi-0.5.9.tgz#7e1c3eaa4b5cc5680072c66b59622c2cb5a7bfff"
  integrity sha512-B9wL+1J3J/9sVqj3DMJtTYuBSx0QdFFu6QwPlw2zVvB995lWivG2vABTYwoWuyUbXwcVvSVHEgPyFBRx5zoPqw==
  dependencies:
    "@coinbase/wallet-sdk" "^3.3.0"
    "@wagmi/core" "^0.4.7"
    "@walletconnect/ethereum-provider" "^1.7.8"
    react-query "^4.0.0-beta.23"
    use-sync-external-store "^1.1.0"
```

**Comment by jxom** [comment]
we are publishing a fix v soon :)

**Comment by tmm** [comment]
Fixed in `0.5.10`

**Comment by oveddan** [comment]
thank you for the very fast turnaround!!!!

**Comment by DefiCake** [comment]
Is v4 still maintained? I have the same issue but I see it did not get any fix. Should I migrate?

**Comment by jxom** [comment]
This was fixed in `0.5.10`.
